.github/workflows/testing-and-deployment.yml

```diff
@@ -190,7 +198,15 @@ jobs:
         if: ${{ matrix.vtk-version == '9.1' || matrix.vtk-version == '9.0.3' }}
         run: pip install 'matplotlib<3.6'
 
+      - name: Set up numpy 2.0
+        if: ${{ matrix.numpy-version != 'default' }}
+        run: |
+          pip uninstall numpy matplotlib -y
+          pip install --pre --no-deps -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple matplotlib
+          pip install --pre --no-deps numpy==2.0.0b1
```

Comment:
You are right, we didn't need the `--pre` when directly pinning.  I removed the pin however so that we get the latest rc candidate.  We now need the `--pre`.

Using `>` doesn't work with pre releases.  Since pip does not require a hierarchy here, it doesn't know whether `2.0.0b1` is higher or lower than `2.0.0rc1`.  I'd argue it should be able to compare  within a single kind of pre release, but this doesn't seem possible.

Reply:
This machinery needs to be updated when 2.0 is fully released in any case here.  Given the current failures with our CI with no clear way to fix them, we will have problems at that point.